app/Http/Controllers/Admin/AdminController.php

```diff
@@ -34,4 +42,58 @@ public function toggleBlock($id)
         }
         return back()->with('update-status-failed', trans('message.update-status-failed'));
     }
+
+    public function blogList(Request $request)
+    {
+        $auth = auth()->user();
+        $blogs = $this->blogService->blogList($request->only('query', 'category_id'));
+        $categories = $this->categoryService->getCategories();
+        return view("admin.blog-list", compact('blogs', 'auth', 'categories'));
+    }
+
+    public function blogEdit($id)
+    {
+        $auth = auth()->user();
+        $blog = $this->blogService->getBlog($id);
+        $categories = $this->categoryService->getCategories();
+        return view("admin.blog-detail", compact('blog', 'categories', 'auth'));
+    }
+
+    public function blogUpdate($id, Request $request)
+    {
+        if ($request->file('file')) {
+            $uploadFile = $this->uploadFileService->uploadFile($request->file('file'));
+            $request = array_merge($request->only('category_id', 'title', 'content'), ['image' => $uploadFile]);
+        } else {
+            $request = $request->only('category_id', 'title', 'content');
+        }
+        $blog = $this->blogService->getBlog($id);
+        $updateBlog = $this->blogService->update($id, $request);
+        if ($updateBlog) {
+            $this->uploadFileService->deleteFile($blog->image);
+            return back()->with('blog-update-success', trans('message.update-success'));
+        }
+        return back()->with('blog-update-failed', trans('message.update-failed'));
+    }
+
+    public function toggleApproved($id)
+    {
+        $toggleApproved = $this->blogService->toggleApproved($id);
+        if ($toggleApproved) {
+            return back()->with('blog-update-success', trans('message.update-success'));
+        }
+        return back()->with('blog-update-failed', trans('message.update-failed'));
+    }
+
+    public function blogDelete($id)
```

Comment:
sửa lại tất tên function
```suggestion
    public function delete($id)
```